lib/modules/manager/tekton/index.ts

```diff
@@ -0,0 +1,120 @@
+import is from '@sindresorhus/is';
+import { loadAll } from 'js-yaml';
+import { ProgrammingLanguage } from '../../../constants';
+import { logger } from '../../../logger';
+import { DockerDatasource } from '../../datasource/docker';
+import type { PackageDependency, PackageFile } from '../types';
+
+export const language = ProgrammingLanguage.Docker;
+
+export const defaultConfig = {
+  // Tekton uses YAML files to define its kubernetes resources. These
+  // don't have a specific file name pattern. To avoid issues with
+  // unrelated YAML files, the match pattern is left empty by default.
+  // Users must provide this value based on their own Tekton usage.
+  // For example, to match all the YAML files in a repository, add
+  // the following to renovate.json in the corresponding git repo:
+  //  {
+  //    "tekton": {
+  //      "fileMatch": ['\\.yaml$', '\\.yml$']
+  //    }
+  //  }
+  fileMatch: [],
+};
+
+export const supportedDatasources = [DockerDatasource.id];
+
+export function extractPackageFile(
+  content: string,
+  fileName: string
+): PackageFile | null {
+  logger.trace('tekton.extractPackageFile()');
+  const deps: PackageDependency[] = [];
+  let docs: any[];
+  try {
+    docs = loadAll(content) as any[];
```

Comment:
please create proper typescript interfaces

Reply:
Unfortunately, I don't have a full set of all the possible types the yaml document could represent. There is also no common set of attributes between those documents that satisfy this.

The required attributes could be found in the PipelineRun and TaskRun resources. These could be defined in their own yaml document, or could be nested under the TriggerTemplate resource. I can create 3 or 4 different Typescript interfaces to cover these.

There's also the case where they could be defined across multiple yaml documents, and files, when using something like [kustomize](https://github.com/kubernetes-sigs/kustomize), in which case we'll lose some flexibility if using typescript interfaces (I think).

Initially, this seemed like a good case to **not** use a typescript interface, but I'm happy to change it if that's more inline with the project. 

Given this information, would you still like me to change the code to use typescript interfaces?

Reply:
you don't need to define all possible interfaces and properties, just the required ones so we've more type safety. 

i don't think we loose any flexibility. it's just more work.

Reply:
Very well. Updated the code to do so.